modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/message/QueryStartRequest.java

```diff
@@ -54,14 +57,24 @@
 
     /** */
     @SuppressWarnings("AssignmentOrReturnOfFieldWithMutableType")
-    public QueryStartRequest(UUID qryId, String schema, String root, AffinityTopologyVersion ver,
-        FragmentDescription fragmentDesc, Object[] params) {
+    public QueryStartRequest(
+        UUID qryId,
+        String schema,
+        String root,
+        AffinityTopologyVersion ver,
+        FragmentDescription fragmentDesc,
+        int totalFragmentsCnt,
+        Object[] params,
+        byte[] paramsBytes
+    ) {
         this.qryId = qryId;
         this.schema = schema;
         this.root = root;
         this.ver = ver;
         this.fragmentDesc = fragmentDesc;
+        this.totalFragmentsCnt = totalFragmentsCnt;
         this.params = params;
+        this.paramsBytes = paramsBytes;
```

Comment:
if i correctly understand this optimization helps get rid of multiple serialization of equal data ? if it so - code become complicated to understand ..

Reply:
I've changed comments a little bit, hope it's more clear now.